packages/ee/ui-platform/src/lib/components/upgrades/upgrades.component.ts

```diff
@@ -0,0 +1,68 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { FlagService, fadeInUp400ms } from '@activepieces/ui/common';
+import { Observable, map, combineLatest } from 'rxjs';
+import { PlatformSettingsBaseComponent } from '../platform-settings-base.component';
+import { ApFlagId } from '@activepieces/shared';
+import { AsyncPipe } from '@angular/common';
+
+const compareVersions = (latestVersion: string, currentVersion: string) => {
+  const latest = latestVersion.split('.').map(Number);
+  const current = currentVersion.split('.').map(Number);
+
+  if (latest[1] > current[1]) {
```

Comment:
This is great but currently we are under < 1.0.0 but this will break in future.

The best thing to do is use `semver` package

```typscript
const compareVersions = (
  latestVersion: string,
  currentVersion: string
): VersionComparisonResult => {
  if (semver.gt(latestVersion, currentVersion)) {
    const diff = semver.diff(latestVersion, currentVersion);
    let message = '';
    if (diff === 'minor'  || diff == 'major') {
      message = 'Major update is available. Please update.';
    } else if (diff === 'patch') {
      message = 'Patch update is available. Please update.';
    }
    return {
      emoji: '😞',
      message: message,
    };
  }

  return {
    emoji: '🙂',
    message: 'Up to date!',
  };
};
```

something along these line

Reply:
Yes! Makes sense. Will work on the change now.